worker-workflow/src/main/resources/workflow-control.js

```diff
@@ -371,17 +371,34 @@ function isEmptyMap(mapValue) {
     return jsonString === '{}';
 }
 
+function searchFieldsForValue(fieldValues, value) {
```

Comment:
@dermot-hardy am I ok to have refactored existing fieldHasStringValue function and brought out the actual value comparison so it can also be used for fieldHasAnyStringValue?
Added couple of tests to ensure fieldHasStringValue still works as expected